File: Java/SpectableJavaExamplesTest/src/test/java/spectable/common/ResultString.java
package spectable.common;

import java.util.Objects;
import production.*;
import records.*;
import calculator.*;

public class ResultString {
    private static final String DNCString = "?DNC?";

    public String addressMatches;

    public ResultString(String addressMatches) {
        this.addressMatches = addressMatches;
    }

    /** Builds from the text form, e.g. Money as "25 USD". */
    public static ResultString fromText(String text) {
        java.util.List<String> parts = Tokens.require(text, 1, "Result");
        return new ResultString(parts.get(0));
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof ResultString)) return false;
        ResultString that = (ResultString) o;
        return (DNCString.equals(addressMatches) || DNCString.equals(that.addressMatches) || Objects.equals(addressMatches, that.addressMatches));
    }

    @Override
    public int hashCode() {
        return Objects.hash(addressMatches);
    }

    @Override
    public String toString() {
        return Tokens.token(addressMatches);
    }
}

File: CSharp/SpecStudioExamplesTestsCSharp/common/ResultString.cs
namespace SpecStudioExamplesTestsCSharp.common
{
using production;

    public class ResultString
    {
        public string addressMatches;

        public ResultString(string addressMatches)
        {
            this.addressMatches = addressMatches;
        }

        /// <summary>Builds from the text form, e.g. Money as "25 USD".</summary>
        public static ResultString FromText(string text)
        {
            var parts = Tokens.Require(text, 1, "Result");
            return new ResultString(parts[0]);
        }

        public ResultTyped ToResultTyped()
        {
            return new ResultTyped(
                new List<MatchTyped>()
            );
        }

        public override string ToString()
        {
            return Tokens.Token(addressMatches);
        }

        const string DNCString = "?DNC?";

        public override bool Equals(object? obj)
        {
            if (obj is not ResultString other) return false;
            return (DNCString == this.addressMatches || DNCString == other.addressMatches || object.Equals(this.addressMatches, other.addressMatches));
        }

        public override int GetHashCode()
        {
            var h = new System.HashCode();
            h.Add(this.addressMatches);
            return h.ToHashCode();
        }
    }
}
